Content.Server/Chemistry/EntitySystems/ChemistrySystemHypospray.cs

```diff
@@ -30,23 +38,101 @@ private void OnSolutionChange(EntityUid uid, HyposprayComponent component, Solut
             Dirty(component);
         }
 
-        public void OnAfterInteract(EntityUid uid, HyposprayComponent comp, AfterInteractEvent args)
+        public void OnAfterInteract(EntityUid uid, HyposprayComponent component, AfterInteractEvent args)
         {
             if (!args.CanReach)
                 return;
 
             var target = args.Target;
             var user = args.User;
 
-            comp.TryDoInject(target, user);
+            TryDoInject(component, target, user);
         }
 
-        public void OnAttack(EntityUid uid, HyposprayComponent comp, MeleeHitEvent args)
+        public void OnAttack(EntityUid uid, HyposprayComponent component, MeleeHitEvent args)
         {
             if (!args.HitEntities.Any())
                 return;
 
-            comp.TryDoInject(args.HitEntities.First(), args.User);
+            TryDoInject(component, args.HitEntities.First(), args.User);
+        }
+
+        public bool TryDoInject(HyposprayComponent component, EntityUid? target, EntityUid user)
+        {
+            if (!EligibleEntity(target, _entMan))
+                return false;
+
+            string? msgFormat = null;
+
+            if (target == user)
+            {
+                msgFormat = "hypospray-component-inject-self-message";
+            }
```

Comment:
```suggestion
            if (target == user)
                msgFormat = "hypospray-component-inject-self-message";
```

Just consistency here since it's just a one line if.